flixel/system/FlxBasePreloader.hx

```diff
@@ -193,51 +231,143 @@ class FlxBasePreloader extends NMEPreloader
 		return bmpData;
 		#end
 	}
-	
+
 	/**
 	 * Site-locking Functionality
 	 */
 	private function checkSiteLock():Void
 	{
 		#if web
-		if (!_urlChecked && (allowedURLs != null))
+		if (_urlChecked) return;
+
+		if (!isHostUrlAllowed())
 		{
-			if (!isHostUrlAllowed())
-			{
-				var tmp = new Bitmap(new BitmapData(stage.stageWidth, stage.stageHeight, true, FlxColor.WHITE));
-				addChild(tmp);
-				
-				var format = new TextFormat();
-				format.color = 0x000000;
-				format.size = 16;
-				format.align = TextFormatAlign.CENTER;
-				format.bold = true;
-				format.font = "system";
-				
-				var textField = new TextField();
-				textField.width = tmp.width - 16;
-				textField.height = tmp.height - 16;
-				textField.y = 8;
-				textField.multiline = true;
-				textField.wordWrap = true;
-				textField.defaultTextFormat = format;
-				textField.text = "Hi there!  It looks like somebody copied this game without my permission.  Just click anywhere, or copy-paste this URL into your browser.\n\n" + allowedURLs[0] + "\n\nto play the game at my site.  Thanks, and have fun!";
-				addChild(textField);
-				
-				textField.addEventListener(MouseEvent.CLICK, goToMyURL);
-				tmp.addEventListener(MouseEvent.CLICK, goToMyURL);
-				
-				removeEventListener(Event.ENTER_FRAME, onEnterFrame);
-			}
-			else
-			{
-				_urlChecked = true;
-			}
+			removeChildren();
+			removeEventListener(Event.ENTER_FRAME, onEnterFrame);
+
+			createSiteLockFailureScreen();
+		}
+		else
+		{
+			_urlChecked = true;
 		}
 		#end
 	}
-	
+
 	#if web
+	/**
+	 * When overridden, allows the customized creation of the sitelock failure screen.
+	 */
+	private function createSiteLockFailureScreen():Void
+	{
+		addChild(createSiteLockFailureBackground(0xffffff, 0xe5e5e5));
+		addChild(createSiteLockFailureIcon(0xe5e5e5, 0.9));
+		addChild(createSiteLockFailureText(30));
+	}
+
+	private function createSiteLockFailureBackground(innerColor:UInt, outerColor:UInt):Shape
```

Comment:
Not sure.  I fairly certain `FlxColor` would work.  I most likely wrote it this way since the underlying graphics APIs expect `UInt`.